File: 6_kyu/tribonacci_sequence.py
def tribonacci(signature, n):
    x = [a for a in signature ]
    for i in range(2,n-1):
        y = x[i]
        y += x[i-1] + x[i-2]
        x.append(y)
    return x if n>3 else [signature[j] for j in range(n)]

tribonacci([1,2,3], 10)

File: Dart/6_kyu/tribonacci_sequence.dart
List<num> tribonacci(List<num> signature, int n) {
  List<num> x = List.from(signature);
  for (int i = 2; i < n - 1; i++) {
    num y = x[i] + x[i - 1] + x[i - 2];
    x.add(y);
  }
  return n > 3 ? x : signature.sublist(0, n);
}